spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindHiddenMethod.java

```diff
@@ -0,0 +1,143 @@
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
+import org.apache.bcel.Const;
+import org.apache.bcel.generic.Type;
+
+import java.util.Arrays;
+
+/**
+ * This detector finds all the methods of a subclass which are hiding the static methods of the superclass.
+ * Please see @see <a href="https://wiki.sei.cmu.edu/confluence/display/java/MET07-J.+Never+declare+a+class+method+that+hides+a+method+declared+in+a+superclass+or+superinterface">SEI CERT MET07-J</a>
+ * @author Nazir, Muhammad Zafar Iqbal
+ */
+public class FindHiddenMethod implements Detector {
+    private final BugReporter bugReporter;
+
+    public FindHiddenMethod(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+    /*
+     * Only the super classes are analyzed, not the super interfaces
+     * Reason: when I looked in detail of the super interfaces,
+     * I found there is no point to analyze as whenever we have method hiding due to super interfaces,
+     * We can call the method by fully qualified name only (as Interface can't be instantiated).
+     * Calling a hidden method by fully qualified name is a compliant test.
+     * See @see <a href="https://wiki.sei.cmu.edu/confluence/display/java/MET07-J.+Never+declare+a+class+method+that+hides+a+method+declared+in+a+superclass+or+superinterface">SEI CERT MET07-J</a>
+     * Thereof, it is useless to analyze super interfaces in the detector.
+     */
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        //First, I get the subclass and superclasses in variables.
+        //This is the current class. Named it subClass to better depict the idea of sub and super class.
+        JavaClass subClass = classContext.getJavaClass();
+        //No need for null check as every class has at least `Object` as its super class.
+        JavaClass[] superClasses;
+        try {
+            superClasses = subClass.getSuperClasses();
+        } catch (ClassNotFoundException e) {
+            AnalysisContext.reportMissingClass(e);
+            return;
+        }
+        //I store all the methods of subclass in an array
+        Method[] methods = subClass.getMethods();
+        //I filter the non-private and static methods, as private methods can't be overridden.
+        //Then for each method, I go through each method of the superclass for checking method hiding.
+        for (Method method : methods) {
+            //Careful!!! regarding the order of the conditions applied here
+            //Taking advantage of short circuit evaluation here by placing !method.isStatic() first.
+            //I check for the exceptional cases of inner class using three auxiliary private methods.
+            if (!method.isStatic() || method.isPrivate() || isMainMethod(method) ||
+                    isConstructor(method) || isHidingInnerClass(method) || isAutoGeneratedMethod(method)) {
```

Comment:
Can you please modify the formatting like this, so it is conform with the codebase?

```suggestion
            if (!method.isStatic() || method.isPrivate() || isMainMethod(method)
                    || isConstructor(method) || isHidingInnerClass(method) || isAutoGeneratedMethod(method)) {
```